src/utils/config_utils/src/config_utils.c

```diff
@@ -15,11 +15,48 @@
 #include <azure_c_shared_utility/strings_types.h>
 #include <parson.h>
 #include <parson_json_utils.h>
+#include <pthread.h>
 #include <stdbool.h>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
 
+static pthread_mutex_t s_config_mutex = PTHREAD_MUTEX_INITIALIZER;
+
+static ADUC_ConfigInfo s_configInfo = { 0 };
+
+static inline void s_config_lock(void)
+{
+    pthread_mutex_lock(&s_config_mutex);
+}
+
+static inline void s_config_unlock(void)
+{
+    pthread_mutex_unlock(&s_config_mutex);
+}
+
+//static const char* CONFIG_LOG_FOLDER = "logFolder";
+static const char* CONFIG_ADU_SHELL_FOLDER = "aduShellFolder";
```

Comment:
Do these need to be static? We could make these #defines at the top for more readability/utility. Probably a nit but wanted to suggest. 

Also commented out CONFIG_LOG_FOLDER probably needs to be removed.